src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -38,6 +38,33 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
                 .anyMatch(preppedWord::equalsIgnoreCase);
     }
 
+    /**
+     * Returns true if any word in the {@code sentence} starts the {@code word}.
+     *   Ignores case, and a substring match suffices.
+     *   <br>examples:<pre>
+     *       containsStartSubstringIgnoreCase("ABc def", "ab") == true
+     *       containsStartSubstringIgnoreCase("ABc def", "de") == true
+     *       containsStartSubstringIgnoreCase("ABc def", "ef") == false //not the starting part of the word.
+     *       </pre>
+     * @param sentence cannot be null
+     * @param word cannot be null, cannot be empty, must be a single word
+     */
+    public static boolean containsStartSubstringIgnoreCase(String sentence, String word) {
+        requireNonNull(sentence);
+        requireNonNull(word);
+
+        String preppedWord = word.trim();
+        checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
+        checkArgument(preppedWord.split("\\s+").length == 1, "Word parameter should be a single word");
```

Comment:
does that mean we can't search for Software Engineer?

Reply:
Good point! I have resolved this